### DISCUSSION TITLE
Why is Relevance score is always = 0?

### DISCUSSION THREAD

**Discussion by Vasiliy-b** [Cinnamon/kotaemon#390] [Q&A]
Despite the LLM providing the correct response, the documents listed in the informational panel always contain irrelevant retrievals, and their scores are always 0. Why is this happening, any guess? 
*Using hybrid search

![image](https://github.com/user-attachments/assets/707e33cd-8f21-402d-a8c1-eba733d00c7d)

**Comment by Vasiliy-b** [comment]
Also, it seems that while using the 'complex' reasoning settings, the model is also using web search. This is not clear to me—when does it have access to web search and when does it not? Is this related to hybrid search? I thought hybrid search combines text and vector search.

Any hints on how this works? When I decompose the request, there is no sign of the model using tools like web search...
![image](https://github.com/user-attachments/assets/5dbc3fc2-f977-4758-a66d-05b496a74d60)

`Chosen rewrite pipeline DecomposeQuestionPipeline(
  (llm): ChatOpenAI(api_key=sk-DmivWJqglpxu..., base_url=https://api.ope..., frequency_penalty=None, logit_bias=None, logprobs=None, max_retries=None, max_retries_=2, max_tokens=None, model=gpt-4o, n=1, organization=None, presence_penalty=None, stop=None, temperature=None, timeout=20, tool_choice=None, tools=None, top_logprobs=None, top_p=None)`

**Comment by Vasiliy-b** [comment]
After adding cohere API key, relevance score is calculating propperly. 

![image](https://github.com/user-attachments/assets/892450c0-dfad-49ba-811a-c5bcc8dfcca5)


But, strange thing with web search are still there. Does model uses tools by default?

**Reply by Vasiliy-b** [reply]
Seems like when model fail to find relevant answer in docs it is going web by default.

![image](https://github.com/user-attachments/assets/2fa5e814-3519-45f3-b94b-8d875d5af194)

**Reply by vap0rtranz** [reply]
> But, strange thing with web search are still there. Does model uses tools by default?

Not all models support tools[], so even if Kotaemon requests a tool, the model could reject it.

Look for errors from Kotaemon app.  I usually forget to switch to a model that supports tools[].  The UI does not tell you when the model fails tools[].

**Comment by vap0rtranz** [comment]
My relevance scores report non-zero when only using my local vectorstore.

![image](https://github.com/user-attachments/assets/b77808d4-80b0-4170-97fe-c9d2a3566145)

Notice that my re-ranking score is the only 0.  

It seems the settings that funnel into the relevance scores are scatter about the UI.  For example, the checkbox for "Use re-ranking" under Settings->File Collection will attempt to call Cohere.  I found that about because I saw an error in the app logs.
